pkg/provisioner/controller.go

```diff
@@ -99,6 +100,12 @@ func NewController(provisionerName string, provisioner api.Provisioner, clientse
 			if newObc.ObjectMeta.DeletionTimestamp != nil && oldObc.ObjectMeta.DeletionTimestamp != nil {
 				return
 			}
+			if !reflect.DeepEqual(newObc.Spec, oldObc.Spec) {
+				if reflect.DeepEqual(newObc.Spec.AdditionalConfig, oldObc.Spec.AdditionalConfig) {
+					log.Error(nil, "Only Additional Config can be modified for an OBC")
```

Comment:
@jeffvance : will this suffice as an error check for modifying other fields apart from `additionalconfig`, so that update happens specifically.

Reply:
The `if` in [L100](https://github.com/kube-object-storage/lib-bucket-provisioner/pull/213/files#diff-1e346e3430c7a1bec9a1fde398367f33b921671c6c3ced081ad42e4d8e345522R100) is checking that if _both_ the old and new OBCs are being deleted then `return`.
I think you only need to check if the new OBC is being deleted and then return.

Reply:
> will this suffice as an error check for modifying other fields apart from additionalconfig?

Yes. I think I missed seeing the first DeepEqual on spec.

Reply:
Go error messages are supposed to not start with capital letters. I also suggest rewording for clarity.

```suggestion
					log.Error(nil, "invalid changes to OBC. only additionalConfig can be updated after creation")
```

Reply:
I read that this code checks 
```go
if new spec has changed {
  if new additional config is the same as old additional config {
    logError
    return
    // this part seems correct to me
  }
}
```

But what if a user modifies both the spec and the additionalConfig?

I think the check needs to be more like:
```go
// create copy of old spec, and set the new spec's additionalConfig on it
oldSpecWithNewAdditionalConfig := oldObc.Spec.DeepCopy()
oldSpecWithNewAdditinoalConfig.AdditionalConfig = newObc.Spec.AdditionalConfig.DeepCopy()

if !reflect.DeepEqual(newObc.Spec, oldObc.Spec) {
    // new OBC is changed
    if !reflect.DeepEqual(oldSpecWithNewAdditionalConfig, newObc.Spec) {
        // new OBC has changed something other than additionalConfig
        log.Error(nil, "invalid changes to OBC. only additionalConfig can be updated after creation")
        return
    }
}
```